src/parser/jobs/blm/modules/RotationWatchdog.tsx

```diff
@@ -0,0 +1,563 @@
+import {t} from '@lingui/macro'
+import {Plural, Trans} from '@lingui/react'
+import _ from 'lodash'
+import React, {Fragment} from 'react'
+import {Icon, Message} from 'semantic-ui-react'
+
+import {ActionLink, StatusLink} from 'components/ui/DbLink'
+import {RotationTable} from 'components/ui/RotationTable'
+import {getDataBy} from 'data'
+import ACTIONS from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import {CastEvent, Event} from 'fflogs'
+import Module, {dependency} from 'parser/core/Module'
+import Checklist, {Requirement, Rule} from 'parser/core/modules/Checklist'
+import Combatants from 'parser/core/modules/Combatants'
+import Enemies from 'parser/core/modules/Enemies'
+import Invulnerability from 'parser/core/modules/Invulnerability'
+import Suggestions, {SEVERITY, Suggestion, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import Timeline from 'parser/core/modules/Timeline'
+
+import DISPLAY_ORDER from './DISPLAY_ORDER'
+import {FIRE_SPELLS} from './Elements'
+import {BLM_GAUGE_EVENT} from './Gauge'
+
+const DEBUG_SHOW_ALL_CYCLES = false && process.env.NODE_ENV !== 'production'
+
+const EXPECTED_FIRE4 = 6
+const NO_UH_OPENER_FIRE4 = 5
+const FIRE4_FROM_MANAFONT = 1
+
+const MIN_MP_FOR_FULL_ROTATION = 9600
+const THUNDERCLOUD_MILLIS = 18000
+
+const AFUIBUFFMAXSTACK = 3
+
+const ISSUE_SEVERITY_TIERS = {
+	1: SEVERITY.MINOR,
+	3: SEVERITY.MEDIUM,
+	5: SEVERITY.MAJOR,
+}
+
+const CYCLE_ENDPOINTS = [
+	ACTIONS.BLIZZARD_III.id,
+	ACTIONS.TRANSPOSE.id,
+	ACTIONS.FREEZE.id,
+]
+
+// This is feelycraft at the moment. Rotations shorter than this won't be processed for errors.
+const MIN_ROTATION_LENGTH = 3
+
+/**
+ * Error type codes, higher values indicate higher priority errors. If you add more, adjust the IDs to ensure correct priorities.
+ * Only the highest priority error will be displayed in the 'Reason' column.
+ * NOTE: Cycles with values below ERROR_CODES.SHORT will be filtered out of the RotationTable display
+ * unless the DEBUG_SHOW_ALL_CYCLES variable is set to true
+ */
+const ERROR_CODES = {
+	NONE: 0,
+	FINAL_OR_DOWNTIME: 1,
+	SHORT: 2,
+	MISSING_FIRE4S: 10, // These two errors are lower priority since they can be determined by looking at the
+	MISSING_DESPAIRS: 15, // target columns in the table, so we want to tell players about other errors first
+	MANAFONT_BEFORE_DESPAIR: 30,
+	EXTRA_T3: 49, // Extra T3 and Extra F1 are *very* similar in terms of per-GCD potency loss
+	EXTRA_F1: 50, // These two codes should stay close to each other
+	NO_FIRE_SPELLS: 75,
+	DROPPED_ENOCHIAN: 100,
+}
+
+/**
+ * If you add an error code above, give it a message here too, even ones filtered out normally
+ * so they show something in debug mode.
+ */
+const ERROR_MESSAGES = {
```

Comment:
It's not _too_ bad with only the defs above and the messages here, so not going to block on it right now, but consider combining it a-la

```js
const ERRORS = {
	NONE: {priority: 0, message: 'No Errors'},
	// ...
}
```